---
title: "Assignment6"
author: "Sonali Joshi"
date: "4/22/2017"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(GenomicRanges)
library(SummarizedExperiment)
library(GenomicFeatures)
library(DESeq2)
library(ggplot2)
library(dplyr)
library(ggrepel)

```

## Exploratory Analysis
The Counts data and Phenotype data were loaded into R for analysis with Bioconductor.
Rows with very low counts were filtered out to reduce the size of the counts data.
```{r GetData}
CountsData<-(read.delim('FeatureCountsData.txt', header = TRUE, row.names="Genes"))
CountsData = CountsData[rowMeans(CountsData)>5,]
CountsData = as.matrix(CountsData)
head(CountsData,5)
PhenoData<-as.matrix(read.delim('Sample_phenotypes.txt',header=TRUE))
sample_data <- as.matrix(PhenoData)
head(PhenoData)

```
The PCA done during exploratory analysis showed clustering by age. Hence, the hypothesis is that the genes are differentially expresses by age. The null hypothesis is that there is no difference in gene expression between the fetal and adult brain tissue.

##Create DESeqDataSet object

The package DESeq2 was used for further analysis. A column called ExpDesign was added to the Phenotype table to label the fetal and adult samples. This is used by DESeq2 package to specify the experimental design for analysis. 
All samples have the same race, while there are four female and two male samples. Sex was adjusted as a covariate in the analysis.

Construct the DESeqDataSet object from the Counts data and the Phenotype data

```{r }
data_dds <- DESeqDataSetFromMatrix(CountsData, PhenoData, ~ExpDesign + Sex)
head(data_dds)
```

## Significance testing

```{r }
data_ddsSE <- DESeq(data_dds)
head(data_ddsSE)
```

## Direction of the fold change
```{r }
data_results <- results(data_ddsSE,contrast=c("ExpDesign","Fetal","Adult"), alpha = 0.05)
mcols(data_results, use.names = T)
summary(data_results)
```

## Dataframe of p-values and fold change

Use the adjusted p-value as it is corrected for multiple comparisons. As the values are small take the -log10 of the adjust p-value to better visualize the magnitude.

```{r }
data_values <- data.frame(gene = row.names(CountsData), 
                          pvalue = data_results$pvalue, 
                          padj = data_results$padj, 
                          log10_adj_pvalue = -log10(data_results$padj), 
                          logfc = data_results$log2FoldChange)

data_values <- na.omit(data_values)

sorted_data_values <- data_values[order(-data_values$log10_adj_pvalue),]
head(sorted_data_values,10)

#generate the required tab delimited file
write.table(sorted_data_values, file = "sorted_data_values.txt", sep = "\t", 
            row.names = F)

#Find number of differentially expressed genes with adjusted p value < 0.001
num_diffex_genes <- subset(sorted_data_values, padj <= 0.001)
dim(sorted_data_values)
dim(num_diffex_genes)
```


## Make a Volcano plot

```{r }
Vplot1 <- ggplot(data_values, aes(x=logfc, y = log10_adj_pvalue)) + 
  geom_point(size = 3, alpha = 0.7, na.rm = T) +
  ggtitle(label = "Volcano Plot") + # Add a title
  xlab(expression(log[2]("Fetal" / "Adult"))) + # x-axis label
  ylab(expression(-log[10]("adjusted p-value"))) + # y-axis label
  geom_vline(xintercept = 0, colour = "black")# + # Add 0 lines

Vplot1

```


``` {r }
#Genes up regulated
up_genes <- data_values %>% filter(logfc > 1) %>% arrange (padj)
head(up_genes)
up_gene_list <- (as.matrix(up_genes$gene))

#Genes down regulated
down_genes <- data_values %>% filter(logfc < -1 ) %>% arrange (padj)
down_gene_list <- (as.matrix(down_genes$gene))
head(down_genes)

```

